### PR TITLE
[WEB-3426] 15.0.0 - Implement V2 Buttons

### PR DESCRIPTION
**Pull request by jamiehenson**
## Jira Ticket Link / Motivation

In order to advance with the `ably-ui` update in docs, we need to bite the bullet and get the buttons in ably-ui updated, which are sorely out of date. From here, we can build the new docs stuff with the new buttons in place.

## Summary of changes

Existing button styles have been reorganised into a legacy designation so as to minimise visual regressions from this point. New styles have been added with a new convention of `.ui-button-{VARIANT}-{SIZE}`, and a `Button` React component has also been added for a more fluid experience when developing in Voltaire/Docs (it's a somewhat thin wrapper around a styled `button` element).

Other misc changes:
- I've reorganised the sections in Storybook from "CSS", "Features", "JS Components", "Brand" to "Features", "Components", "Styles" to better reflect the new structure of the components. Makes the PR quite noisy but nothing has changed with the components.

With this, I think it's time to bump the major version - so much has changed lately, both in terms of code changes and philosophy, and this along with the (currently unreleased) theming work points to a justifiable jump above 14.

## How do you manually test this?

[Crack open a good book
](https://ably.github.io/ably-ui/?path=/docs/components-button--docs)
## Merge/Deploy Checklist

- [ ] Written automated tests for implemented features/fixed bugs
- [ ] Rebased and squashed commits
- [ ] Commits have clear descriptions of their changes
- [ ] Checked for any performance regressions

## Frontend Checklist

- [ ] No frontend changes in this PR
- [ ] Added before/after screenshots for changes
- [ ] Tested on different platforms/browsers with [Browserstack](https://www.browserstack.com)
- [ ] Compared with the initial design / our [brand guidelines](https://www.figma.com/file/jTQgyIovrhGBOf4Zrvjvbk/Ably-Linear-Design?node-id=118%3A0)
- [ ] Checked the code for accessibility issues ([VoiceOver User Guide](https://support.apple.com/en-gb/guide/voiceover/welcome/mac))?

## Release Notes

- **New Features**
	- Introduced a new `Button` component with customizable variants and sizes.
	- Added a new `searchDataId` prop to the `Meganav` component.
	- Enhanced dark theme styles for improved visual consistency.

- **Improvements**
	- Streamlined story organization by updating component titles from "JS Components" to "Components" across multiple stories.
	- Updated button styles for better modularity and consistency.

- **Bug Fixes**
	- Adjusted checkbox rendering to ensure the initial state is correctly defined.

- **Documentation**
	- Improved documentation for button variants and sizes, as well as tooltip interactions.

### CODE REVIEW

File: src/core/styles/colors/Colors.stories.tsx [hunk 1]
```diff
@@ -2,15 +2,12 @@ import React from "react";
...
+  title: "Styles/Colors",
 };
 
 const colorSet = (colors, useClass = "") =>
   colors.map((color) => (
-    <div
-      key={color}
-      className="rounded-lg w-128 ui-shadow-lg-soft flex flex-col"
-    >
+    <div key={color} className="rounded-lg w-128 bg-neutral-100 flex flex-col">
```
[the remark this answers is not in the document]
Comment (jamiehenson) [reply src/core/styles/colors/Colors.stories.tsx:10]: Good suggestion, I've implemented this.

File: src/core/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  return (
+    <button
+      className={clsx(
+        buttonClasses[variant][size ?? "md"],
+        { [iconModifierClasses[size ?? "md"].left]: leftIcon },
+        { [iconModifierClasses[size ?? "md"].right]: rightIcon },
+        className,
+      )}
+      {...rest}
+    >
```
[the remark this answers is not in the document]
Comment (jamiehenson) [reply src/core/Button.tsx:83]: Good instinct, but these values would be incorrect.

File: src/core/styles/legacy-buttons.css [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    @apply text-mid-grey bg-gui-unavailable cursor-not-allowed pointer-events-none select-none;
+  }
+
+  .ui-btn-secondary.ui-btn-disabled {
+    @apply text-gui-unavailable border-gui-unavailable bg-white cursor-not-allowed pointer-events-none select-none;
+  }
+
+  .ui-btn-secondary-invert.ui-btn-disabled {
+    @apply text-gui-unavailable border-gui-unavailable cursor-not-allowed pointer-events-none select-none;
+  }
```
[the remark this answers is not in the document]
Comment (jamiehenson) [reply src/core/styles/legacy-buttons.css:94]: We're leaving the legacy classes alone for now in order to focus on the new buttons.

File: src/core/styles/buttons.css [hunk 4]
```diff
@@ -1,95 +1,109 @@
...
+    @apply ui-button-base ui-gui-button3 h-[40px] px-20 py-[10px] gap-8 [&>svg]:!w-20 [&>svg]:!h-20;
+  }
+
+  .ui-button-xs-base {
+    @apply ui-button-base ui-gui-button4 h-[36px] px-12 py-[9px] gap-6 [&>svg]:!w-16 [&>svg]:!h-16;
+  }
+
+  .ui-button-disabled-base {
+    @apply disabled:bg-gui-unavailable dark:disabled:bg-gui-unavailable-dark disabled:text-gui-unavailable-dark disabled:dark:text-gui-unavailable disabled:hover:bg-gui-unavailable dark:disabled:hover:bg-gui-unavailable-dark;
+  }
```
[the remark this answers is not in the document]
Comment (jamiehenson) [reply src/core/styles/buttons.css:24]: Good bot

**Review by kennethkalmer — COMMENTED** [review]
I like what you did here, and bumping the major version is also a good call for this. The new button naming convention makes sense and instantly feels familiar which is nice. I'll leave the final ✅ for @aralovelace.

File: src/core/Button.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+// got to go the long way round because of ol' mate Taily Waily
+const buttonClasses: Record<ButtonType, Record<ButtonSize, string>> = {
+  priority: {
+    lg: "ui-button-priority-lg",
+    md: "ui-button-priority",
+    sm: "ui-button-priority-sm",
+    xs: "ui-button-priority-xs",
+  },
+  primary: {
```
Comment (aralovelace) [inline src/core/Button.tsx:41]: love these!

**Review by aralovelace — APPROVED** [review]
looks good! Cant wait to use these new buttons!

**Pull request merged**